serialization/src/extras/non_empty_vec.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization_core::{Decode, Encode};
+
+/// A wrapper that saves the extra byte coming from Option by assuming the Vec can never be empty
+///
+/// Description:
+/// - If the Vec has data, it encodes to just the Vec, the Option is omitted
+/// - If the Vec has no data, it encodes to None
+/// - Some(vec![]) and None are equivalent when encoded, but when decoded result in None
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct DataOrNoVec<T>(pub Option<Vec<T>>);
```

Comment:
But it's not OptionVec ;-)

Reply:
I would argue that, because `DataOrNo` literally means optional but in a weird way. If you imply that it is a "optimised encoder-decoder for option vector" then yes, but current name imho is confusing. Also I guess it would be used as an `Option` via `.0` in the code.
Or did I miss your point?

Reply:
Unfortunately there's no better name that incorporates the fact that this is a "data in a vector otherwise None". If you call this OptionVec, then it sounds like an `Option<Vec>`, which it's not quite the same. It's something special. I'm all open for better naming, and believe me I renamed it like 3 times before creating the PR... but I can't figure out a better name that I like and makes sense and doesn't take 87 characters.

Reply:
I won't insist on this. `OptionVec` would be my choice, but it should not block this PR